### PR TITLE
Jetpack connect: Use authorize schema

### PR DESCRIPTION
**Pull request by sirreal**
The PR restores persistence schema usage for `jetpackConnectAuthorize`. It was previously unused and simply injected cached state.

## Testing
1. Test should tell whether the schema definition is valid (they saved me 😁 )
1. Go through the connection flow refreshing frequently during the authorize step.
1. Verify that valid states are persisted (watch the console)

### CODE REVIEW

File: client/state/jetpack-connect/reducer/schema.js [hunk 1]
```diff
@@ -1,26 +1,31 @@
...
-	type: 'object',
-	additionalProperties: false,
-	patternProperties: {
-		'^.+$': {
+	anyOf: [
+		{
+			type: 'object',
+			additionalProperties: false,
+			properties: {},
+		},
```
Comment (sirreal) [inline client/state/jetpack-connect/reducer/schema.js]: If schema validation fails or the timestamp is stale, `{}` is used. Allow matching empty object.

File: client/state/jetpack-connect/reducer/schema.js [hunk 2]
```diff
@@ -1,26 +1,31 @@
...
+		{
 			type: 'object',
+			additionalProperties: false,
 			required: [ 'timestamp' ],
 			properties: {
-				authorizationCode: { type: 'string ' },
```
Comment (sirreal) [inline client/state/jetpack-connect/reducer/schema.js]: See that ugly space in there? New unit tests caught this one 😎

Comment (seear) [reply client/state/jetpack-connect/reducer/schema.js]: 🎉

**Comment by sirreal** [comment]
Seeing schema validation errors with `additionalProperties`. Investigating…

**Pull request merged**